tree/tree/src/TTree.cxx

```diff
@@ -4802,6 +4814,53 @@ Int_t TTree::FlushBaskets() const
    Int_t nerror = 0;
    TObjArray *lb = const_cast<TTree*>(this)->GetListOfBranches();
    Int_t nb = lb->GetEntriesFast();
+
+#ifdef R__USE_IMT
+   if (ROOT::IsImplicitMTEnabled() && fIMTEnabled) {
+      if (fSortedBranches.empty()) { const_cast<TTree*>(this)->InitializeSortedBranches(); }
+
+      BoolRAIIToggle sentry(fIMTFlush);
+      fIMTZipBytes.store(0);
+      fIMTTotBytes.store(0);
+      std::atomic<Int_t> nerrpar(0);
+      std::atomic<Int_t> nbpar(0);
+      std::atomic<Int_t> pos(0);
+      tbb::task_group g;
```

Comment:
Certainly!

The point is that working code with IMT-disabled may stop working when IMT is enabled.  One would hope that issues would be fixable once the user understands the underlying issue.  Hence the extensive comment in the code.